services/cd-service/pkg/repository/transformer.go

```diff
@@ -1620,6 +1624,109 @@ func generateReleaseTrainResponse(envDeployedMsg, envSkippedMsg map[string]strin
 	return resp
 }
 
+type Overview struct {
+	App     string
+	Version uint64
+}
+
+func getEnvironmentInGroup(groups []*api.EnvironmentGroup, groupNameToReturn string, envNameToReturn string) *api.Environment {
+	for _, currentGroup := range groups {
+		if currentGroup.EnvironmentGroupName == groupNameToReturn {
+			for _, currentEnv := range currentGroup.Environments {
+				if currentEnv.Name == envNameToReturn {
+					return currentEnv
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getOverrideVersions(commitHash, upstreamEnvName string, repo Repository) (resp []Overview, err error) {
+	oid, err := git.NewOid(commitHash)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating new oid for commitHash %s: %w", commitHash, err)
+	}
+	s, err := repo.StateAt(oid)
+	if err != nil {
+		var gerr *git.GitError
+		if errors.As(err, &gerr) {
+			if gerr.Code == git.ErrNotFound {
+				return nil, fmt.Errorf("ErrNotFound: %w", err)
+			}
+		}
+		return nil, fmt.Errorf("unable to get oid: %w", err)
+	}
+
+	if envs, err := s.GetEnvironmentConfigs(); err != nil {
+		return nil, fmt.Errorf("unable to get EnvironmentConfigs for %s: %w", commitHash, err)
+	} else {
+		result := mapper.MapEnvironmentsToGroups(envs)
+		for envName, config := range envs {
+			var groupName = mapper.DeriveGroupName(config, envName)
+			var envInGroup = getEnvironmentInGroup(result, groupName, envName)
+			if upstreamEnvName != envInGroup.Name || upstreamEnvName != groupName {
+				continue
+			}
+			if apps, err := s.GetEnvironmentApplications(envName); err != nil {
+				return nil, fmt.Errorf("unable to get EnvironmentApplication for env %s: %w", envName, err)
+			} else {
+				for _, appName := range apps {
+					app := api.Environment_Application{
+						Name: appName,
+					}
+					var version *uint64
+					if version, err = s.GetEnvironmentApplicationVersion(envName, appName); err != nil && !errors.Is(err, os.ErrNotExist) {
+						return nil, fmt.Errorf("unable to get EnvironmentApplicationVersion for %s: %w", appName, err)
+					} else {
+						if version == nil {
+							continue
+						} else {
+							app.Version = *version
+						}
+					}
+					resp = append(resp, Overview{App: app.Name, Version: app.Version})
+				}
+			}
+		}
+	}
+	return resp, nil
```

Comment:
this is maybe a nitpick, but we should stay consistent with indentation style/early exits.
Instead of
```
if envs, err := foo(); err != nil {
  return ...
} else { ...
```

Applying this here will save 3 indentation levels.
Please do
```
envs, err := foo()
if err != nil {
  return ...
} // no else here
 ...
```

Reply:
will do, thanks for noticing that!